ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -99,10 +101,16 @@
    *                            that gets created as part of this poll operation.
    */
   void poll(RequestRegistrationCallback<DeleteOperation> requestRegistrationCallback) {
-    cleanupExpiredInflightRequests();
-    checkAndMaybeComplete();
-    if (!isOperationComplete()) {
-      fetchRequests(requestRegistrationCallback);
+    try {
+      cleanupExpiredInflightRequests();
+      checkAndMaybeComplete();
+      if (!isOperationComplete()) {
+        fetchRequests(requestRegistrationCallback);
+      }
+    } catch (Exception e) {
+      routerMetrics.unexpectedErrorCount.inc();
```

Comment:
did you consider pros and cons of doing this at the operation level v/s doing this try-catch at a higher level i.e. at the `RequestResponseHandler` thread itself?